src/pages/use-sequence.ts

```diff
@@ -16,18 +16,26 @@ const editFlowSequence = [
   PageRoute.LPA,
   PageRoute.Title,
   PageRoute.Description,
-  // missing pages
+  PageRoute.UpdateTimetableStatus,
+  PageRoute.StatusChangeEvent,
   PageRoute.PublishLocalDevelopmentScheme,
   ...stages.map((stage) => stage.key),
   PageRoute.Export,
 ];
 
-export const useSequence = (userJourney: Journey | null) => {
+export const useSequence = (
+  userJourney: Journey | null,
+  timetableStatusHasChanged?: boolean
+) => {
   const navigate = useNavigate();
   const { pathname } = useLocation() as { pathname: PageRoute };
 
   const sequence =
-    userJourney == Journey.Create ? createFlowSequence : editFlowSequence;
+    userJourney == Journey.Create
+      ? createFlowSequence
+      : timetableStatusHasChanged
+      ? editFlowSequence
+      : editFlowSequence.filter((page) => page !== PageRoute.StatusChangeEvent);
```

Comment:
I don't love the readability of nested ternaries (especially the way Prettier decides to format them). I was previously wondering if we could compose the sequence:

```ts
const routeOnCondition = (route: PageRoute, condition: boolean) =>
  condition ? [route] : [];

const getSequence = (
  userJourney: Journey | null,
  timetableStatusHasChanged?: boolean
) => {
  // Does this make sense? We'll be redirected to the root if userJourney isn't defined anyway
  if (!userJourney) {
    return [];
  }

  return [
    ...routeOnCondition(
      PageRoute.UploadTimetable,
      userJourney === Journey.Update
    ),
    PageRoute.LPA,
    PageRoute.Title,
    PageRoute.Description,
    ...routeOnCondition(
      PageRoute.UpdateTimetableStatus,
      userJourney === Journey.Update
    ),
    ...routeOnCondition(
      PageRoute.StatusChangeEvent,
      userJourney === Journey.Update && !!timetableStatusHasChanged
    ),
    PageRoute.PublishLocalDevelopmentScheme,
    ...stages.map((stage) => stage.key),
    PageRoute.Export,
  ];
};
```

And then in the hook, we would do something like:

```ts
const sequence = useMemo(
  () => getSequence(userJourney, timetableStatusHasChanged),
  [userJourney, timetableStatusHasChanged]
);
```

This way, we're conditionally adding routes without the need to duplicate the shared ones.

Reply:
I like this solution! Thanks Ronan 🙂

Reply:
I think returning [] makes sense if Journey is null. I ll leave that in 👍

Reply:
Awesome, sounds good